test/core/test_lwt_stream.ml

```diff
@@ -31,6 +31,17 @@ let suite = suite "lwt_stream" [
        t3 >>= fun x3 ->
        return ([x1; x2; x3] = [1; 1; 1]));
 
+  test "of_seq"
+    (fun () ->
+       let seq = fun () -> Seq.Cons (1, Seq.empty) in
```

Comment:
To test "The elements are evaluated from [s] and pushed onto the stream as the stream is consumed," I suggest to cause side effects (setting a ref) on evaluation of either the element 1 or `Nil`, check the side effects are not triggered at the beginning, and are triggered after the corresponding `Lwt_stream.get` call.

Reply:
Sorry for the looong delay on this!  69a1f2b should implement something like you're talking about.